README.md

```diff
@@ -1,2 +1,266 @@
-# lb4-log-extension-example
+# loopback4-example-log-extension
 An example repo showing how to write a complex log extension
+
+## Overview
+
+This repository shows you how to use [loopback4-extension-starter]() to write a complex logging extension that requires a [Component](), [Decorator](), and a [Mixin]().
+
+To use the extension, the user would load the component to get access to a `LogFn` that can be used in a sequence to log information. A Mixin allows the user to set the application wide logLevel. Only Controller methods configured at or above the logLevel will be logged. 
+
+Possible levels are: DEBUG < INFO < WARN < ERROR < OFF
+
+*Possible levels are represented as numbers but users can use `LOG_LEVEL.${level}` to specify the value instead of using numbers.*
+
+A decorator is provided to allow a user to provide metadata for Controller methods by using it to set the minimum logLevel the method should be logged at.
+
+### Example Usage
+```ts
+import {LogLevelMixin, LogComponent, LOG_LEVEL, log} from 'loopback4-example-log-extension';
+// Other imports ... 
+
+class LogApp extends LogLevelMixin(Application) {
+  constructor() {
+    super({
+      components: [RestComponent, LogComponent],
+      logLevel: LOG_LEVEL.ERROR,
+      controllers: [MyController]
+    });
+  };
+}
+
+class MyController {
+  @log(LOG_LEVEL.WARN)
+  @get('/')
+  hello() {
+    return 'Hello LoopBack';
+  }
+
+  @log(LOG_LEVEL.ERROR)
+  @get('/name')
+  helloName() {
+    return 'Hello Name'
+  }
+}
+```
+
+## Tutorial
+
+We'll start by cloning [loopback4-extension-starter]() and updating files headers to be `loopback4-example-log-extension` and `package.json`.
+
+We'll be keeping all the top level files but modify the `src` directory and `test` directory as follows.
+
+### `/src/index.ts`
+
+We'll update the exports accordingly here.
```

Comment:
Remove all of the uses of "We'll ..." below and use the imperative instead, e.g. "Start by defining ... "

Reply:
Would it be better if we list what the user needs to export?